src/hotspot/share/oops/cpCache.cpp

```diff
@@ -309,6 +309,7 @@ ResolvedMethodEntry* ConstantPoolCache::set_method_handle(int method_index, cons
 
   // Store appendix, if any.
   if (has_appendix) {
+    assert(method_entry->has_resolved_references_index(), "sanity");
```

Comment:
The assert is not necessary here, as resolved_references_index() already has the same assert.